Applications/Spire/Include/Spire/Ui/DropDownList.hpp

```diff
@@ -0,0 +1,139 @@
+#ifndef SPIRE_DROP_DOWN_LIST_HPP
+#define SPIRE_DROP_DOWN_LIST_HPP
+#include <QVBoxLayout>
+#include "Spire/Spire/Spire.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/DropDownItem.hpp"
+#include "Spire/Ui/DropDownWindow.hpp"
+#include "Spire/Ui/ScrollArea.hpp"
+
+namespace Spire {
+
+  //! Represents a pop-up drop down list.
+  class DropDownList : public DropDownWindow {
+    public:
+
+      //! Signals that an item was highlighted using the keyboard.
+      /*!
+        \param value The item's value.
+      */
+      using ActivatedSignal = Signal<void (const QVariant& value)>;
+
+      //! Signals that an item was highlighted using the mouse.
+      /*!
+        \param value The item's value.
+      */
+      using HighlightedSignal = Signal<void (const QVariant& value)>;
+
+      //! Signals that an item was selected.
+      /*!
+        \param index The item's list index.
+      */
+      using IndexSelectedSignal = Signal<void (int index)>;
+
+      //! Signals that an item was selected.
+      /*!
+        \param value The item's value.
+      */
+      using ValueSelectedSignal = Signal<void (const QVariant& value)>;
+
+      //! Constructs a DropDownList.
+      /*!
+        \param items The items to display.
+        \param is_click_activated True if the DropDownList should be shown
+                                  or hidden in response to parent widget
+                                  mouse/keyboard activation events, false
+                                  otherwise.
```

Comment:
Instead of saying "True if X, false otherwise." use the convention "True iff X." as per:

https://en.wikipedia.org/wiki/If_and_only_if